src/Files.App/Helpers/SoftwareHelpers.cs

```diff
@@ -11,16 +11,34 @@ public static bool IsVSCodeInstalled()
 		{
 			string registryKey = @"Software\Microsoft\Windows\CurrentVersion\Uninstall";
 
-			var key = Registry.CurrentUser.OpenSubKey(registryKey);
+			return
+				ContainsName(Registry.CurrentUser.OpenSubKey(registryKey), "Microsoft Visual Studio Code") ||
+				ContainsName(Registry.LocalMachine.OpenSubKey(registryKey), "Microsoft Visual Studio Code");
+		}
+
+		public static bool IsVSInstalled()
+		{
+			string registryKey = @"SOFTWARE\Microsoft\VisualStudio";
+
+			var key = Registry.LocalMachine.OpenSubKey(registryKey);
 			if (key is null)
 				return false;
 
-			string? displayName;
+			key.Close();
+
+			return true;
+		}
+
+		private static bool ContainsName(RegistryKey? key, string find)
+		{
+			if (key is null)
+				return false;
 
+			string? displayName;
```

Comment:
declaration of `displayName` could be moved into the foreach loop

Reply:
Won't that impact performance? It would declare a new variable at each cycle. I know it wouldn't make that much difference

Reply:
I am pretty sure that it doesn't

if you look at the lowered code you will see that there is no difference, example  [here](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBAGABAlAFgG5NMEBmfJXAYVwG9dnNnnLdgB7LgG1wAqAUwDOAFxQAKADJRxAHgLYAfLgDWQgJ4iANLgDKYgE6wA5rgBmsACYBKVmwaYAkK4tcjQiAGMAFrkkANwgjXBEAV2AAaS1cWHUtEXsMV2cnFNSlAH5cazkAB14ITQA5CDAhXABeMMiYzQA6aSEYUzFfBoEuQxNWyVtSDNcoCwCAQiUGgEkREvDeXgB5IwB1XygxIX18nyFJPJFC4rKK21wAMnPcgqLS8qEGwxCxERWN30krGDtk1LSXP4IADsuGM4SEgz+AF8Ac4YRhYcDLBBeCIIQ5mPC2LgMfgqNw+IJRGIkDI5GJFCgVAltHoemZLDZktj0pkqTkDkc7hVIW4PF4/AFgqEItFYvENNpfq5WalObcTpUaqL6k0Wm0Ol16X0BrDnCNxpMZnMFss1hstjtvHt5cd7mdLtdDgr7o8xM9Xu9PkzpTK9c4kWD0UM4bD4YiQRYUWjBtj4VCgA=)